lib/hexpm/web/templates/dashboard/key/index.html.eex

```diff
@@ -30,11 +36,38 @@
 
       <div class="panel-body-part">
         <%= form_for @changeset, Routes.key_path(Endpoint, :create), [method: :post], fn f -> %>
-          <%= label f, :name, "Generate API key" %>
+          <%= label f, :name, "Key name" %>
           <div class="form-group">
             <%= text_input f, :name, class: "form-control", placeholder: "Name" %>
             <%= error_tag f, :name %>
           </div>
+          <%= inputs_for f, :permissions, fn f -> %>
+            <%= label f, :domain, "Permission domain" %>
+            <div class="form-group">
+              <%=
+                select(
+                  f,
+                  :domain,
+                  [
+                    API: "api",
+                    "Single organization": "repository",
+                    "All organizations": "repositories"
+                  ],
+                  class: "form-control",
+                  placeholder: "Domain"
+                )
+              %>
+              <%= error_tag f, :domain %>
+            </div>
+            <%= label f, :domain, "Domain resource" %>
+            <div class="form-group">
+              <%= text_input f, :resource, class: "form-control", placeholder: "Resource" %>
+              <%= error_tag f, :resource %>
+              <span id="name-helpbox" class="help-block">
+                Empty, "read", or "write" for API keys. Organization name for single organization.
```

Comment:
wdyt about replacing input and helpbox with a select that limits the options?

```html
<select>
  <optgroup label="API:">
  <option>None (TODO: this applies to API and resources domains?)</option>
  <option>Read</option>
  <option>Write</option>

  <optgroup label="Repository:">
  <option>Org1</option>
  <option>Org2</option>
</select>
```

when I'm back from vacation I can look into adding some JS sprinkles to adjust options dynamically in separate PR

Reply:
I will play the optgroups, that's a good idea. Javascript would be even better so I would appreciate that, but we should try to make sure it works without javascript also.